examples/watering/main.c

```diff
@@ -0,0 +1,196 @@
+                    set_and_notify(valve.active, HOMEKIT_UINT8(0));
+                }
+            }
+        }
+        vTaskDelay(pdMS_TO_TICKS(1000));
+    }
+}
+
+void start_stop_task(bool start) {
+    if(start == true && task_running == false) {
+        task_running = true;
+        vTaskResume(gate_task);
+    } 
+    if(start == false) {
+        bool all_off = true;
+        for (int i=0; i < valve_count; i++) {
+            valve_t valve = valves[i];
+            if(valve.active->value.int_value == 1 && valve.in_use->value.int_value == 1) {
+                all_off = false;
+            }
+        }
+        if(all_off == true && task_running == true) {
+            task_running = false;
+            vTaskSuspend(gate_task);
+        }
+    }
+}
+
+void update_running(bool target, valve_t *thisValve) {
+    if(target == true) {
+        update_valve_state(true, thisValve);
+        set_and_notify(thisValve->remaining_duration, thisValve->set_duration->value);
+        start_stop_task(true);
+    } else {
+        update_valve_state(false, thisValve);
+        start_stop_task(false);
+    }
+}
+
+void on_update_active(homekit_characteristic_t *ch, homekit_value_t value, void *context) {
+    valve_t *thisValve = (valve_t *) context;
+    if(value.int_value == 1) {
+        update_running(true, thisValve);
+    } else {
+        update_running(false, thisValve);
+    }
+}
+
+/* relay controllers */
+void update_valve_state(bool state, valve_t *thisValve) {
+    if(state) {
+        printf("--> open gate number: %d \n", thisValve->gpio);
+        set_and_notify(thisValve->in_use, HOMEKIT_UINT8(1));
+    }
+    else { 
+        printf("--> close gate number: %d \n", thisValve->gpio);
+        set_and_notify(thisValve->in_use, HOMEKIT_UINT8(0));
+    }
+}
+
+void valve_identify(homekit_value_t value) {
+    printf("Valve identify\n");
+}
+
+void init_accessory() {
+    homekit_service_t* services[MAX_SERVICES + 1];
+    homekit_service_t** s = services;
+
+    *(s++) = NEW_HOMEKIT_SERVICE(ACCESSORY_INFORMATION, .characteristics=(homekit_characteristic_t*[]) {
+        NEW_HOMEKIT_CHARACTERISTIC(NAME, "ValveSys1"),
+        NEW_HOMEKIT_CHARACTERISTIC(MANUFACTURER, "buschco"),
+        NEW_HOMEKIT_CHARACTERISTIC(SERIAL_NUMBER, "123"),
+        NEW_HOMEKIT_CHARACTERISTIC(MODEL, "Valve System 1"),
+        NEW_HOMEKIT_CHARACTERISTIC(FIRMWARE_REVISION, "0.1"),
+        NEW_HOMEKIT_CHARACTERISTIC(IDENTIFY, valve_identify),
+        NULL
+    });
+
+    for (int i=0; i < valve_count; i++) {
+        int valve_name_len = snprintf(NULL, 0, "Valve %d", i + 1);
+        char *valve_name_value = malloc(valve_name_len+1);
+        snprintf(valve_name_value, valve_name_len+1, "Valve %d", i + 1);
+        valves[i].number = i;
+        valves[i].gpio = valve_gpios[i];
+        homekit_service_t *nextService = NEW_HOMEKIT_SERVICE(VALVE, .characteristics=(homekit_characteristic_t*[]) {
+            NEW_HOMEKIT_CHARACTERISTIC(NAME, valve_name_value),
```

Comment:
All the valves will have `ValveSys1` as their name. I can not explain why 😢

Reply:
Ok I solved it by adding the `SERVICE_LABEL_NAMESPACE` service. Each valve gets the `SERVICE_LABEL_INDEX`. Each Service will have an unique name: _Zone 1, Zone 2...._